PopupMenu/PopupMenuState.js

```diff
@@ -0,0 +1,92 @@
+import FloatingLayer from '@enact/ui/FloatingLayer';
+import {forward} from '@enact/core/handle';
+import hoc from '@enact/core/hoc';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const forwardHide = forward('onHide');
+
+const PopupState = hoc((config, Wrapped) => {	// eslint-disable-line no-unused-vars
+	return class extends React.Component {
+		static displayName = 'PopupState'
+
+		static propTypes = {
+			noAnimation: PropTypes.bool,
+			noAutoDismiss: PropTypes.bool,
+			onClose: PropTypes.func,
+			open: PropTypes.bool,
+			scrimType: PropTypes.oneOf(['transparent', 'translucent', 'none'])
+		}
+
+		static defaultProps = {
+			noAnimation: false,
+			noAutoDismiss: false,
+			open: false,
+			scrimType: 'translucent'
+		}
+
+		constructor (props) {
+			super(props);
+			this.state = {
+				floatLayerOpen: this.props.open,
+				popupOpen: this.props.noAnimation
+			};
+		}
+
+		componentWillReceiveProps (nextProps) {
+			if (!this.props.open && nextProps.open) {
+				this.setState({
+					popupOpen: nextProps.noAnimation,
+					floatLayerOpen: true
+				});
+			} else if (this.props.open && !nextProps.open) {
+				this.setState({
+					popupOpen: nextProps.noAnimation,
+					floatLayerOpen: !nextProps.noAnimation
```

Comment:
Ouch, this makes my brain hurt!  Why isn't it just `false` for `floatLayerOpen`?

Reply:
Oof! That whole file got unintentionally committed. It's not even referenced!